src/app/dynamic-forms/fields/checkbox/checkbox.component.html

```diff
@@ -1,15 +1,17 @@
-<div *ngIf="(field.cinchyColumn.dataType == 'Yes/No' && field.cinchyColumn.canView)">
+<div *ngIf="(field.cinchyColumn.dataType === 'Yes/No' && field.cinchyColumn.canView)">
   <fa-icon [icon]="faCheckSquare"></fa-icon>
   &nbsp;
-  <input class="m-tb-10 mr-5" type='checkbox' [(ngModel)]="field.value" [ngModelOptions]="{standalone: true}"
-         [disabled]="(field.cinchyColumn.canEdit=== false || field.cinchyColumn.isViewOnly || isDisabled)"
-         (change)="valueChanged()" [id]="field.label" />
+  <input class="m-tb-10 mr-5" type="checkbox"
+         [(ngModel)]="value" [ngModelOptions]="{ standalone: true }"
+         [disabled]="!canEdit"
+         [id]="field.label"
+         (change)="valueChanged()" />
   <div class="checkbox-field">
 
     &nbsp;
-    <label class="pre-formatted" [for]="field.label" [title]="field.caption ? field.caption : ''">
+    <label class="pre-formatted" [for]="field.label" [title]="field.caption || ''">
       {{field.label}}
-      {{field.cinchyColumn.isMandatory == true && (field.value == '' || field.value == null) ? '*' : ''}}
+      {{(field.cinchyColumn.isMandatory && !field.value) ? '*' : ''}}
```

Comment:
Would this make it trigger if the checkbox has a false value? I think that's why the initial implementation only checked for null or empty string